framework/src/modules/chain/loader.js

```diff
@@ -17,28 +17,22 @@
 const async = require('async');
```

Comment:
I don't understand how moving different modules of chain to one level up make any difference. Same you can do but keeping the files where these belongs to.

Reply:
For more clarity of dependency tree, but a lot more coming afterwards